### PR TITLE
parse team.invite links in team sigchain

### PR DESCRIPTION
**Pull request by maxtaco**
- sanity check the invites
- inform the TeamSigChainState object about any changes, but....
- no tests for now; to help patrick compile his PR
- look for bugfixes and tests in an upcoming PR

### CODE REVIEW

File: go/teams/chain_parse.go [hunk 1]
```diff
@@ -37,6 +39,19 @@ type SCTeamMembers struct {
 	None    *[]SCTeamMember `json:"none,omitempty"`
 }
 
+type SCTeamInvites struct {
```
Comment (mlsteele) [inline go/teams/chain_parse.go]: I don't understand why you would want this format. It seems like it makes the checks complicated. What about just a list of `SCTeamInvite`s which are

```
type SCTeamInvite struct {
    ID keybase1.TeamInviteID
    Type string
    Name string
    Role keybase1.TeamRole
}
```

Comment (maxtaco) [inline go/teams/chain_parse.go]: This is what's the server wants. It's done this way to be consistent with the `members`

File: protocol/avdl/keybase1/teams.avdl [hunk 2]
```diff
@@ -120,6 +123,32 @@ protocol teams {
...
 
+  enum TypeInviteCategory {
+    NONE_0,
+    UNKNOWN_1,
+    KEYBASE_2,
+    EMAIL_3,
+    SBS_4
+  }
+
+  variant TeamInviteType switch (TypeInviteCategory c) {
```
Comment (mlsteele) [inline protocol/avdl/keybase1/teams.avdl]: What is type for when there's category too?

Comment (maxtaco) [inline protocol/avdl/keybase1/teams.avdl]: `SBS > twitter`, `SBS > github`, etc...

File: go/teams/chain.go [hunk 3]
```diff
@@ -957,6 +1004,87 @@ func (t *TeamSigChainPlayer) inflateLinkHelper(
...
+
+	for _, invite := range all {
+		res, err := invite.i.TeamInvite(invite.role)
+		if err != nil {
+			return nil, nil, err
+		}
+		id := res.Id
+		_, seen := byID[id]
+		if seen {
+			return nil, nil, NewInviteError(fmt.Sprintf("Invite ID %s appears twice in invite set", id))
```
Comment (mlsteele) [inline go/teams/chain.go]: This checks within the link, but not whether it collides with previous links. So they will just get overridden. This probably won't cause any problems, but we should decide on the rules and at least have the server enforce them.

**Comment by keybase-ci-visdiff** [comment]
The commits f157610eb14f...c703e4f18192 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499359389400-f157610eb14f-c703e4f18192-linux/Markdown-emoji-f157610eb14f.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499359389400-f157610eb14f-c703e4f18192-linux/Markdown-emoji-c703e4f18192.png)  
   Markdown-emoji comparison
</summary>

**Pull request merged**